analytics-core/src/test/java/com/segment/analytics/messages/BatchTest.java

```diff
@@ -26,6 +28,22 @@ public void create(TestUtils.MessageBuilderFactory factory) {
     assertThat(batch.context()).isEqualTo(context);
   }
 
+  @Test
+  public void createWithSentAt(TestUtils.MessageBuilderFactory factory) throws ParseException {
```

Comment:
better to have another test with messages has `sentAt` as nulls, so we have both paths covered.